sound/soc/sof/intel/byt.c

```diff
@@ -657,8 +661,6 @@ EXPORT_SYMBOL_NS(tng_chip_info, SND_SOC_SOF_MERRIFIELD);
 
 #endif /* CONFIG_SND_SOC_SOF_MERRIFIELD */
 
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_BAYTRAIL)
```

Comment:
I found that the IS_ENABLED() check needs to be put back to avoid
unused-function warnings:

diff --git a/sound/soc/sof/intel/byt.c b/sound/soc/sof/intel/byt.c
index bf871412e6e8..bc6d4ed0375f 100644
--- a/sound/soc/sof/intel/byt.c
+++ b/sound/soc/sof/intel/byt.c
@@ -661,6 +661,7 @@ EXPORT_SYMBOL_NS(tng_chip_info, SND_SOC_SOF_MERRIFIELD);
 
 #endif /* CONFIG_SND_SOC_SOF_MERRIFIELD */
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_BAYTRAIL)
 static void byt_reset_dsp_disable_int(struct snd_sof_dev *sdev)
 {
        /* Disable Interrupt from both sides */
@@ -1066,6 +1067,7 @@ static struct platform_driver snd_sof_acpi_intel_byt_driver = {
        },
 };
 module_platform_driver(snd_sof_acpi_intel_byt_driver);
+#endif /* CONFIG_SND_SOC_SOF_BAYTRAIL */
 
 MODULE_LICENSE("Dual BSD/GPL");
 MODULE_IMPORT_NS(SND_SOC_SOF_INTEL_HIFI_EP_IPC);

Reply:
Thanks @arndb, this is also useful. We should probably clean this up for good. We have shared code between PCI and ACPI drivers, and I should probably move the common code to a dedicated module again. It's not so bad with the EXPORT_SYMBOL_NS, we can declare symbols and make sure they don't pollute the rest of the system.

Would you happen to have this .config so that I can reproduce this faster?

Reply:
There were two failed randconfig builds in my overnight test:
[0xC9DAD133_defconfig.gz](https://github.com/thesofproject/linux/files/5821716/0xC9DAD133_defconfig.gz)
[0xA96BA954_defconfig.gz](https://github.com/thesofproject/linux/files/5821723/0xA96BA954_defconfig.gz)